build-tools/packages/build-cli/src/library/dangerfile.ts

```diff
@@ -47,6 +62,37 @@ export async function dangerfile(): Promise<void> {
 	// there were actual changes to the bundle sizes.  In other cases, we don't post a
 	// message and danger will delete its previous message
 	if (result.comparison === undefined || !bundlesContainNoChanges(result.comparison)) {
+		// Check for bundle size regression
+		let sizeCheck = false;
+		if (result.comparison !== undefined) {
+			for (const bundle of result.comparison) {
+				const totalMetric = bundle.commonBundleMetrics[totalSizeMetricName];
+				const totalParsedSizeDiff =
+					totalMetric.compare.parsedSize - totalMetric.baseline.parsedSize;
+				if (totalParsedSizeDiff > 5120) {
+					sizeCheck = true;
+					break;
+				}
+			}
+		}
+
+		// Warn and add label to PR in case of bundle size regression
+		if (sizeCheck) {
+			warn("Bundle size regression detected -- please investigate before merging!");
+			// Add the label to the PR
+			try {
+				await Promise.resolve(
```

Comment:
Assuming the declared type of createOrAddLabel is correct above, you shouldn't need to wrap this in await/Promise.resolve (can just call it synchronously directly).

Reply:
I decided to keep the labeling. I actually declared the type wrong here, I corrected the declared [createOrAddLabel](https://github.com/danger/danger-js/blob/81645a6a7f37534d156ad495c639c914bf7053e1/source/platforms/github/GitHubUtils.ts#L201) type to be asynchronous.